src/Umbraco.Web.Website/Extensions/UmbracoWebstiteServiceCollectionExtensions.cs

```diff
@@ -21,6 +23,9 @@ public static void AddUmbracoWebsite(this IServiceCollection services)
 
             //TODO figure out if we need more to work on load balanced setups
             services.AddDataProtection();
+
+            // Website security
+            services.AddSingleton<IUmbracoWebsiteSecurity, UmbracoWebsiteSecurity>();
```

Comment:
We will need an Accessor, like we have for BackofficeSecurity. Otherwise the members are shared between sessions

Reply:
I've updated to introduce this if you could please check that it's how you intended.

Reply:
Exactly like this